core/src/main/java/discord4j/core/retriever/StoreEntityRetriever.java

```diff
@@ -104,6 +106,11 @@ public StoreEntityRetriever(GatewayDiscordClient gateway) {
     @Override
     public Flux<Member> getGuildMembers(Snowflake guildId) {
         return stateView.getGuildStore().find(guildId.asLong())
+                .filterWhen(guild -> new Guild(gateway, guild).getVoiceStates()
```

Comment:
It could be interesting to find a way to first check for the intent then create the guild to avoid a maybe unecessary instantation